cmd/kueuectl/app/list/helpers.go

```diff
@@ -77,3 +80,43 @@ func addActiveFilterFlagVar(cmd *cobra.Command, p *[]bool) {
 	cmd.Flags().BoolSliceVar(p, "active", make([]bool, 0),
 		"Filter by active status. Valid values: 'true' and 'false'.")
 }
+
+func addForObjectFlagVar(cmd *cobra.Command, p *string) {
+	cmd.Flags().StringVar(p, "for", "",
+		"Filter workloads to only those pertaining to the specified resource.")
+}
+
+// decodeResourceTypeName handles type/name resource formats and returns a resource tuple
+// (empty or not), whether it successfully found one, and an error
+// copied from https://github.com/kubernetes/kubernetes/blob/8565e375251450a291f0af3c6195c7a5bf890292/staging/src/k8s.io/kubectl/pkg/cmd/events/events.go#L380
+func decodeResourceTypeName(mapper meta.RESTMapper, s string) (gvk schema.GroupVersionKind, name string, found bool, err error) {
+	if !strings.Contains(s, "/") {
+		return
+	}
+	seg := strings.Split(s, "/")
```

Comment:
I understand this is copied but...

```suggestion
	seg := strings.Split(s, "/")
	if len(seg) < 2 {
		return
	}
```
split len if 0 for empty input and 1 if no slash found

Reply:
```suggestion
seg := strings.Split(s, "/")
	if len(seg) > 2 {
		err = fmt.Errorf("arguments in resource/name form may not have more than one slash")
	}
	if len(seg) != 2 {
		return
	}
```
Hmm. Maybe like this?

Reply:
As it is mandatory to have only one slash, just keeping the original condition `len(seg) != 2` is okay. 

Btw, notice that this is outdated.

Reply:
Yes, I saw. But if we have more than 1 slashes we need to return error. Or not?

Reply:
Yes, but we can also reuse the error created here:

https://github.com/kubernetes-sigs/kueue/pull/2238/files#diff-6820a373f0ec6d90ba8d09cf508b433ac48a5613da03d0c27a61dbcc28603f65R182

Reply:
Is it not mistake? If not found it should said resource not found.

Reply:
The `decodeResourceTypeName` func is just parsing the string passed by the user, whether it doesn't contain any slash or contains more than 1, the error is due to an invalid format. So I think the `--for must be in resource/name form` message applies to that case as well.

Reply:
IIUC this function not only parsing but also checking is this resource available on k8s or not.